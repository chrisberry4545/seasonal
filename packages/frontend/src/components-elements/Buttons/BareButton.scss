@import './../../styles/settings';
@import './../../styles/tools';

.c-bare-button {
  border: none;
  cursor: pointer;
  font-family: $primary-font;
}
