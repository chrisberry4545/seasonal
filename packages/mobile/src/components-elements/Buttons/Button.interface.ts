import { StyleProp, TextStyle } from 'react-native';

export interface IButton {
  activeOpacity?: number;
  onClick?: () => void;
  style?: StyleProp<TextStyle>;
}
