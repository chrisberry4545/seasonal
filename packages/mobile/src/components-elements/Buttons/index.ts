export * from './BareButton';

export * from './BorderedButton';
