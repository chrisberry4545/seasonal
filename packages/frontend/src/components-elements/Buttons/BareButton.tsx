import React, { FC } from 'react';
import './BareButton.scss';

export const BareButton: FC<{
  className?: string,
  onClick?: () => void
}> = ({
  className,
  children,
  onClick
}) => (
  <button className={`${(className || '')} c-bare-button`} onClick={onClick}>
    { children }
  </button>
);
