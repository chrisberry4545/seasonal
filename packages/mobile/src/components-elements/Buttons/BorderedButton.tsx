import React, { FC } from 'react';
import { ViewStyle } from 'react-native';

import { BareButton } from './BareButton';
import {
  IButton
} from './Button.interface';
import { colors } from '../../styles/colors';

const styleBorderedButton: ViewStyle = {
  borderColor: colors.greyMed,
  borderWidth: 1,
  padding: 10
};

export const BorderedButton: FC<IButton> = ({
  activeOpacity,
  children,
  onClick,
  style
}) => (
  <BareButton
    style={ { ...styleBorderedButton, ...style } }
    onClick={onClick}
    activeOpacity={activeOpacity}>
    { children }
  </BareButton>
);
