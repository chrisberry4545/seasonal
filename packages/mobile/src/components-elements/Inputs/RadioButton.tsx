import React, { FC } from 'react';
import { BareButton } from '../Buttons';
import { TextMedium } from '../Text';
import { TextStyle } from 'react-native';
import { styles } from '../../styles';

const styleRadioButton: TextStyle = {
  alignItems: 'center',
  display: 'flex',
  flex: 1,
  justifyContent: 'center',
  minHeight: 30
};
const styleRadioButtonSelected: TextStyle = {
  backgroundColor: styles.colors.selectionColor
};

export const RadioButton: FC<{
  isSelected?: boolean,
  label: string,
  onClick: () => void
}> = ({
  isSelected,
  label,
  onClick
}) => (
  <BareButton
    style={{
      ...styleRadioButton,
      ...(isSelected ? styleRadioButtonSelected : {})
    }}
    onClick={onClick}
    >
    <TextMedium>{label}</TextMedium>
  </BareButton>
);
