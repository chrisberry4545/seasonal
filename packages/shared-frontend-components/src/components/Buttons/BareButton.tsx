import React, { FC } from 'react';
import './BareButton.scss';
import { IButton } from './Button.interface';

export const BareButton: FC<IButton> = ({
  className,
  children,
  onClick,
  type = 'button',
  e2eTag
}) => (
  <button
    data-e2e={e2eTag}
    type={type}
    className={`${(className || '')} c-bare-button`}
    onClick={onClick}>
    { children }
  </button>
);
