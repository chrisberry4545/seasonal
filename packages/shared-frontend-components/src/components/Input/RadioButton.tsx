import React, { FC } from 'react';
import './RadioButton.scss';

export const RadioButton: FC<{
  className?: string,
  defaulToChecked?: boolean,
  groupName: string,
  value: string,
  label: string
}> = ({
  className,
  defaulToChecked,
  groupName,
  value,
  label
}) => (
  <label
    htmlFor={value}
    className={`${(className || '')} c-radio-button ${
      defaulToChecked ? 'c-radio-button--selected' : ''
    }`}>
    <input
      className='c-radio-button__input'
      defaultChecked={defaulToChecked}
      id={value}
      type='radio'
      name={groupName}
      value={value} />
    {label}
  </label>
);
