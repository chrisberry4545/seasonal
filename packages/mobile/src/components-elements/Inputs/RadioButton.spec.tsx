import React from 'react';
import { RadioButton } from './RadioButton';
import { shallow, ShallowWrapper } from 'enzyme';
import { BareButton } from '../Buttons';

jest.mock('../Buttons', () => ({
  BareButton: () => 'BareButton'
}));
jest.mock('../Text', () => ({
  TextMedium: () => 'TextMedium'
}));

describe('<RadioButton />', () => {
  let wrapper: ShallowWrapper;
  let mockOnClicked: jest.Mock;

  beforeEach(() => {
    mockOnClicked = jest.fn();
    wrapper = shallow(
      <RadioButton
        isSelected={false}
        label='Radio button test'
        onClick={mockOnClicked} />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('Responds when an option is clicked', () => {
    const onClickProp = wrapper.find(BareButton).props().onClick;
    if (onClickProp) {
      onClickProp();
    }
    expect(mockOnClicked).toHaveBeenCalled();
  });
});
