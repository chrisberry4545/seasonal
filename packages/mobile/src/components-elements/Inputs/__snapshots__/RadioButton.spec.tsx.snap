// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> renders correctly 1`] = `
<BareButton
  onClick={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
      "minHeight": 30,
    }
  }
>
  <TextMedium>
    Radio button test
  </TextMedium>
</BareButton>
`;
